libsdl-org/SDL
# cmake: install SDL2test + add component support + fix mingw

1. Also install SDL2_test when building SDL2 with cmake (you need to configure cmake with `-DSDL_TEST=ON`)
2. Export a `SDL2::SDL2test` target (same name as in #5721)
3. For mingw and when using CMake 3.13+, fix mingw linking with `SDL2::SDL2` by adding `-lmingw32` to the link options of `SDL2::SDL2`.
4. Add `COMPONENTS` support to the `SDL2Config.cmake`, installed by cmake. The name of the targets are the same as in #5721
5. Fixes #2794: When building with MSVC, or targeting Windows/OS2 with Watcom and having enabled `SDL_STATIC`, the static library will always be called `SDL2-static.lib`.
6. Do limited integration testing by using a just-created SDL2 install prefix to build a minimal application
7. Create CMake configuration files for the MinGW redistributable
8. Make the CMake configuration files, generated by autotools, relocatable

The following `CMakeLists.txt` builds fine when using an installation created by cmake.
Download link: [CMakeLists.txt](https://github.com/libsdl-org/SDL/files/8789657/CMakeLists.txt)
Tested using mingw32@Linux + gcc11@Linux

The test cmake script should also work with the VC+Xcode cmake from #5721
